pilot/pkg/kube/inject/mesh.go

```diff
@@ -156,8 +158,14 @@ containers:
   imagePullPolicy: {{ .ImagePullPolicy }}
   {{ end -}}
   resources:
+    {{ "[[ if (isset .ObjectMeta.Annotations $proxyCPUKey) -]]" }}
```

Comment:
Create a variable at the top of the yaml to hold the value of the annotation or it's default:

```yaml
 [[- $proxyCPUValue := (annotation .ObjectMeta $proxyCPUKey "10m") -]]
 [[- $proxyMemoryValue := (annotation .ObjectMeta $proxyMemoryKey "") -]]
```
Then the code here becomes something like:

```yaml
requests:
  cpu:  {{ "\"[[ $proxyCPUValue ]]\"" }}
  {{ "[[ if (ne $proxyMemoryValue '') -]]" }}
  memory: {{ "\"[[ $proxyMemoryValue ]]\"" }}
  {{ "[[- end ]]" }}
```

Reply:
Done.